# HG changeset patch
# User Rik <rik@octave.org>
# Date 1533152523 25200
#      Wed Aug 01 12:42:03 2018 -0700
# Node ID e9f107d31799c9f4a9840a6d13a3ab78712a74b9
# Parent  e3c612c0b7445aca1c1e742002ef4eaea98d16a6
warning_ids.m: Document new Octave:legacy-function warning.

* warning_ids.m: Document new Octave:legacy-function warning.

* setstr.m: Wrap some comments to 80 characters rather than 72.

diff --git a/scripts/help/warning_ids.m b/scripts/help/warning_ids.m
--- a/scripts/help/warning_ids.m
+++ b/scripts/help/warning_ids.m
@@ -207,16 +207,25 @@
 ##
 ## @item Octave:language-extension
 ## Print warnings when using features that are unique to the Octave
 ## language and that may still be missing in @sc{matlab}.
 ## By default, the @code{Octave:language-extension} warning is disabled.
 ## The @option{--traditional} or @option{--braindead} startup options for
 ## Octave may also be of use, @pxref{Command Line Options}.
 ##
+## @item Octave:legacy-function
+## If the @code{Octave:legacy-function} warning is enabled, a
+## warning is issued when Octave encounters a function that Matlab has
+## suggested should be avoided.  The function may become obsolete at some
+## point in the future and removed, in which case the warning will change to
+## @code{Octave:deprecated-function}, and the function will continue to exist
+## for two further versions of Octave before being removed.
+## By default, the @code{Octave:legacy-function} warning is enabled.
+##
 ## @item Octave:logical-conversion
 ## By default, the @code{Octave:logical-conversion} warning is enabled.
 ##
 ## @item Octave:lu:sparse_input
 ## If the @code{Octave:lu:sparse_input} warning is enabled, Octave
 ## will warn when the lu function is called with a sparse input and less than
 ## four output arguments.  In this case, sparsity-preserving column
 ## permutations are not performed and the result may be inaccurate.
diff --git a/scripts/legacy/setstr.m b/scripts/legacy/setstr.m
--- a/scripts/legacy/setstr.m
+++ b/scripts/legacy/setstr.m
@@ -18,20 +18,19 @@
 
 ## -*- texinfo -*-
 ## @deftypefn {Function File} {} setstr (@var{s})
 ## This function is obsolete.  Use char instead.
 ## @end deftypefn
 
 ## Author: jwe
 
-## At one time, Matlab docs stated that this function is obsolete and
-## would be removed in some future version.  Now users are told that it
-## should be avoided, but there is no mention of possible future
-## removal.
+## At one time, Matlab docs stated that this function is obsolete and would be
+## removed in some future version.  Now users are told that it should be
+## avoided, but there is no mention of possible future removal.
 
 function retval = setstr (varargin)
 
   persistent warned = false;
   if (! warned)
     warned = true;
     warning ("Octave:legacy-function",
              "setstr is obsolete; please use char instead");
